server/api.go

```diff
@@ -272,3 +272,111 @@ func (d dexAPI) RevokeRefresh(ctx context.Context, req *api.RevokeRefreshReq) (*
 
 	return &api.RevokeRefreshResp{}, nil
 }
+
+func (d dexAPI) CreateConnector(ctx context.Context, req *api.CreateConnectorReq) (*api.CreateConnectorResp, error) {
+	if req.Connector == nil {
+		return nil, errors.New("no connector supplied")
+	}
+	if req.Connector.Id == "" || req.Connector.Type == "" || req.Connector.Name == "" {
+		return nil, errors.New("Connector ID, Type, and Name are mandatory fields")
+	}
+
+	c := storage.Connector{
+		ID:              req.Connector.Id,
+		Type:            req.Connector.Type,
+		Name:            req.Connector.Name,
+		ResourceVersion: "1",
+		Config:          req.Connector.Config,
+	}
+	if err := d.s.CreateConnector(c); err != nil {
+		if err == storage.ErrAlreadyExists {
+			return &api.CreateConnectorResp{AlreadyExists: true}, nil
+		}
+		d.logger.Errorf("api: failed to create connector: %v", err)
+		return nil, fmt.Errorf("create connector: %v", err)
+	}
+
+	return &api.CreateConnectorResp{}, nil
+}
+
+func (d dexAPI) UpdateConnector(ctx context.Context, req *api.UpdateConnectorReq) (*api.UpdateConnectorResp, error) {
+	if req.Id == "" {
+		return nil, errors.New("no connector ID supplied")
+	}
+	if req.Type == "" && req.Name == "" && len(req.Config) == 0 {
+		return nil, errors.New("nothing to update")
+	}
+
+	updater := func(old storage.Connector) (storage.Connector, error) {
+		if req.Type != "" {
+			old.Type = req.Type
+		}
+
+		if req.Name != "" {
+			old.Name = req.Name
+		}
+
+		if len(req.Config) > 0 {
+			old.Config = req.Config
+		}
+
+		currentVersion, err := strconv.Atoi(old.ResourceVersion)
```

Comment:
which would mean we need to change it in the backend storage too and alter the db tables. I looked at grpc code and they seem to do it the same way. I could switch it all out to int if it seems like a design issue.